language/stdlib/modules/ValidatorConfig.move

```diff
@@ -10,13 +10,8 @@ module ValidatorConfig {
 
     struct Config {
         consensus_pubkey: vector<u8>,
-        /// TODO(philiphayes): restructure
-        ///   3) remove validator_network_identity_pubkey
-        ///   4) remove full_node_network_identity_pubkey
-        validator_network_identity_pubkey: vector<u8>,
-        validator_network_address: vector<u8>,
-        full_node_network_identity_pubkey: vector<u8>,
-        full_node_network_address: vector<u8>,
+        validator_network_addresses: vector<vector<u8>>,
```

Comment:
To your point from slack: if these are fixed length or have a known max length, you could do:

```
struct NetworkAddress { addr: vector<u8> }
public fun create(addr: vector<u8>): NetworkAddress { // add length check }
```

This is somewhat similar to newtypes in that it won't change the data layout of `Config` (i.e., you won't need to change your code in `libra/network`), although it does have the (very small) runtime overhead of creating a `NetworkAddress` struct, doing an extra borrow to get at `addr`, etc.

Reply:
alternatively there is a single entry point for setting that array so a check with abort there seems as good to me. Actually better than a side type, but that is just me...